recommender/recommender.py

```diff
@@ -443,36 +474,35 @@ def add_resource(self, data, _suffix=''):
         for field in self.resource_content_fields:
             result[field] = data[field]
 
+        resource_id = stem_url(data['url'])
         # check url for redundancy
-        for recommendation in self.recommendations:
-            if self.check_redundancy(recommendation['url'], data['url']):
-                result['error'] = ('The resource you are attempting to ' +
-                                   'provide has already existed')
-                for field in self.resource_content_fields:
-                    result['dup_' + field] = recommendation[field]
-                result['dup_id'] = recommendation['id']
-                result['Success'] = False
-                tracker.emit('add_resource', result)
-                return result
-            
+        if resource_id in self.recommendations:
+            result['error'] = ('The resource you are attempting to ' +
+                               'provide has already existed')
+            for field in self.resource_content_fields:
+                result['dup_' + field] = self.recommendations[resource_id][field]
+            result['dup_id'] = self.recommendations[resource_id]['id']
+            result['Success'] = False
+            tracker.emit('add_resource', result)
+            return result
+
         # check url for de-endorsed resources
-        for deendorsed_resource in self.deendorsed_recommendations:
-            if self.check_redundancy(deendorsed_resource['url'], data['url']):
-                result['error'] = ('The resource you are attempting to ' +
-                                   'provide has been de-endorsed by staff, ' +
-                                   'because: ' + deendorsed_resource['reason'])
-                for field in self.resource_content_fields:
-                    result['dup_' + field] = recommendation[field]
-                result['dup_id'] = recommendation['id']
-                result['Success'] = False
-                tracker.emit('add_resource', result)
-                return result
+        if resource_id in self.deendorsed_recommendations:
+            result['error'] = ('The resource you are attempting to ' +
+                               'provide has been removed by staff, ' +
+                               'because: ' + self.deendorsed_recommendations[resource_id]['reason'])
```

Comment:
So, the requested resource is not found? (status code 404 if temporary, 410 if permanent) Or the requested resource exists, but is not being returned for some reason? (status code 403)